### ISSUE TITLE
I should be able to delete the «Fore» and «Back» layers

### ISSUE THREAD

**Issue by ctrlcctrlv** [fontforge/fontforge#3690]
Working on a font with six layers. I should be able to delete the default &laquo;Fore&raquo; and/or &laquo;Back&raquo; layer(s)&mdash;it seems like a weird restriction that I can't.

I can't think of another way (without using a script) to just clear everything on a certain layer.

**Comment by ctrlcctrlv** [comment]
Oh, I just thought of a way without a script. Make a blank layer, then use Edit&rarr;Copy Layer to Layer; that successfully clobbers «Fore».

Still&hellip;this limitation seems so arbitrary and weird.

**Comment by frank-trampe** [comment]
I agree that it would be sort of useful to be able to clear the layers, but I think that there is a lot of code that expects them to exist.

**Comment by skef** [comment]
Yeah, deleting the default layers would cause problems for the reasons @frank-trampe mentions. 

Note that I already implemented this as part of the menu restructuring code. Look for `MID_ClearForeground` and `MID_ClearBackground` in https://github.com/fontforge/fontforge/pull/3638/commits/c9309bb3398b5144ccdaaaf43359887d8e3a0206#diff-085da67dad16e974616022e0856d5b58 . I used `SCClearAll(cv->b.sc, ly_fore);` for the foreground and in testing it seemed to work fine. (I didn't add a check to see if anything was on the layer to grey out the option, as there wasn't one for the background and it seemed like gilding the lily.)

**Comment by skef** [comment]
(I did that because there was/is a "clear" function at the font level, and the principle of that menu design was to have the same menu options across the different windows except in special cases. It seemed silly to have that grayed out in the CharView menu and therefore I added the function. So I guess it's worth mentioning that Edit -> Clear in the font window menu, combined with selecting the relevant glyph(s), already does what you're looking for.)

**Comment by ctrlcctrlv** [comment]
@skef I very much respect you, yet I don't think that's user friendly. The most intuitive place to be able to clear layers would be in the CharView, in the layers list there, and perhaps also in _Font Info&rarr;Layers_.

Why in my font with six layers, should I only be able to clear certain default layers? Why is the only way to clear the ones I created to delete them and recreate them, or use _Copy Layer to Layer_ with a blank layer as input?

**Comment by skef** [comment]
We may have a terminology problem. I'm using "clear" to mean "get rid of all the contents" and "delete" to mean "go looking for the layer and it's not there anymore". This is important because you can add and then delete an *additional* layer, and after that it's just gone, and trying to access it in the `layers` array will give you a `None` or an exception (I haven't checked). 

Allowing the foreground or background of a glyph to be actually *deleted*, as opposed to *cleared*, would pose all sorts of problems.

As far as clearing goes, "clearly" I don't object to being able to do that from the CharView -- I added that code myself in my update to the menu project. It's a good idea! I'm just pointing out that there is an existing, if strangely placed, interface.
